.golangci.yaml

```diff
@@ -0,0 +1,45 @@
+run:
+  timeout: 5m
+linters:
+  enable:
+  - asciicheck
+  - bodyclose
+  - deadcode
+  - depguard
+  - dogsled
+  - durationcheck
+  - errcheck
+  - exhaustive
+  - exportloopref
+  - gochecknoinits
+  - gofmt
+  - goimports
+  - gomnd
+  - gosec
+  - gosimple
+  - govet
+  - importas
+  - ineffassign
+  - megacheck
+  - misspell
+  - nilerr
+  - nolintlint
+  - predeclared
+  - revive
+  - staticcheck
+  - structcheck
+  - typecheck
+  - unconvert
+  - unparam
+  - varcheck
+  - wastedassign
+issues:
+  exclude-rules:
+  - linters:
+    - staticcheck
+    text:  "SA4006" # ignore err not checked in test files
+    path: _test\.go
+  - linters:
+    - ineffassign
+    text:  "ineffectual assignment" # ignore err not checked in test files
+    path: _test\.go
```

Comment:
optional: I'm not sure about this suppression. Ineffassign is a common cause of test bugs imo. Is there a specific reason for this?

Reply:
copypasta from kic

Reply:
removed 1ed276e0262008e18782de603ecffb2998db548f